.github/ISSUE_TEMPLATE/code_quality_issue.yml

```diff
@@ -0,0 +1,46 @@
+name: Code Quality Issue
+description: Create a code quality issue to help Files keep a clean codebase
+labels: [code quality]
+body:
+  - type: textarea
+    attributes:
+      label: Description
+      description: A clear and concise description of what the code quality issue is.
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Concerned code
+      description: A list of the different files and/or areas of the code concerned by the issue.
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Gains
+      description: What would fixing this code quality issue bring to the source code? 
+      value: |
+        - eg. A better readability.
+        - eg. Uncoupling concepts X and Y.
+        - eg. Clarifying the responsability of class C.
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Possibles solutions
+      description: If you have any idea of how to fix the issue, please describe it here. If appropriate, diagrams are welcomed.
```

Comment:
Should this be change to a requirements section?

Reply:
Hmm, unsure.. @QuaintMako

Reply:
I don't think so, because one may not have an idea of how to fix it. For example, one may not know about design patterns or where to use it.

Reply:
My understanding is that the template is primarily aimed at existing contributors familiar with the codebase.

Reply:
Indeed, this kind of issue would be mainly used by contributors with knowledge of the code base. 

But when raising an issue, one may not have studied it enough to know how to fix it. For something needing a lot of work, for example let's say reworking the command process, it may be hard to offer a solution.

Yet it can be argue easily that to raise a code quality issue, one has to see the issue first and seeing an issue usually have an hinch about how to solve it.

So... I don't have a definitive answer.

Reply:
That's good feedback, my recommendation is to include a "Requirements" section that's option. That way the section can be filled in later on when the requirements are finalized.

Reply:
Should the "Requirements" section replace the "Possible Solution" then?

Reply:
Yes

Reply:
Done.